### DISCUSSION TITLE
SDXL - I don't see Refiner's influence when generating an image

### DISCUSSION THREAD

**Discussion by ostap667inbox** [vladmandic/sdnext#1760] [Q&A]
Here are two images with the same Prompt and Seed.
00000 - Generated with Base Model only
00001 - SDXL Refiner model is selected in the "Stable Diffusion refiner" control.

Look at images - they're completely identical. Why did the Refiner model have no effect on the result? What am I missing? When I switch on the Refiner model, do I have to include something else in Settings?

![00000-flying dog with rose flower in his mouth](https://github.com/vladmandic/automatic/assets/123593542/d4b060e8-dda6-4153-9dbb-e7ccfa0c0b17)
![00001-flying dog with rose flower in his mouth](https://github.com/vladmandic/automatic/assets/123593542/68cc0ffa-63db-4f79-bac0-33880915487c)

**Comment by ETA-22** [comment]
Did you enable second pass?

**Reply by ostap667inbox** [reply]
**"Lack of documentation will ruin this project sooner or later"** (c) Nostradamus. :(
Let's go to Wiki: "Second-pass workflows such as hires fix are not yet implemented".
Would I get any ideas to turn on "second pass" and look there?

Whatever. I have not been able to get stable performance on a video card with 12GB VRAM. After the first pass VRAM is full, and when trying to perform the second pass generation stops with the message "WARNING GPU high memory utilisation: 100%". The options "Move base model to CPU when using refiner" and "Move refiner model to CPU when not in use" do not work. -medvram or --lowvram doesn't help much.  After restarting WebUI generation sometimes works, but the next request may stop working again. Everything is unstable and raw now.

**Reply by zethfoxster** [reply]
i had this happening when i had "no half vae" enabled.

**Comment by system1system2** [comment]
I have the same problem (up to commit 57d1d3ed) on macOS. The solution, for me, is to:

1. launch SD.next
2. select None as Stable Diffusion refiner
3. select again the 0.9 refiner

This forces SD.next to load the refiner model and the image looks different from the one generated by the base model alone.

You can tell that the refiner is being used by the fact that the following string appears in the terminal:
`Loading diffuser refiner: stabilityai/stable-diffusion-xl-refiner-0.9`

and by the fact that each image generation is preceeded by 2 progress bars in the terminal.

At this point, you can check/uncheck the `Second Pass` checkbox and SD.next will honor the request to consider/ignore the refiner model.

**If, instead, after launching SD.next, you just check the Second Pass checkbox right away nothing really happens. The refiner model will be ignored.**

**Comment by midcoastal** [comment]
For future readers, something that is also worth noting: Refiners apply only to using Upscalers.  If you are using a `Latent` upscale, the original Model is used to render the upscale, and the Refiner does not trigger.

If you want to have the Refiner trigger, you have to use a non-SD upscaler (like `ESRGAN` or other), and then your Refiner will trigger.

**Reply by ETA-22** [reply]
Interesting! I dont use the refiner generally, but I assumed that the entire point with the refiner in the intended SD workflow was for it to work from the latent image!

**Reply by midcoastal** [reply]
To be fair, that is what I assumed as well.  I ran a lot of testing the other day, and that is best I understand it.

It would seem that, using `backend.DIFFUSERS` anyway, Refiners are by and large not behaving...  I'm awaiting feedback.

**Reply by vladmandic** [reply]
this message is just wrong. refiner do not apply only to upscalers if using latent - that makes no sense.

**Reply by midcoastal** [reply]
> this message is just wrong. refiner do not apply only to upscalers if using latent - that makes no sense.

You are correct, all of my initial assessment was incorrect.  However, after poking for a good few hours, I came to find that you really don't want to upscale *at all*, when using an actual Refiner model.  It goes all pear-shaped...

`stairs, forest`, `Euler`, `20` Steps, No secondary:

![image](https://github.com/vladmandic/automatic/assets/125929259/b56657db-b92a-4661-81fc-86e217b4b219)

Same with Secondary, Upscaler `None`, Refiner Start of `0.5`, Refiner Steps of `10`, Denoise of `0.7`

![image](https://github.com/vladmandic/automatic/assets/125929259/118fb177-5e0b-4c0c-ae34-a9dcd55d9b41)

Same, but with Upscale of `1.5`, using `Nearest`

![image](https://github.com/vladmandic/automatic/assets/125929259/2a76676b-0eb3-4e8d-9c02-ebec48e8cfe3)

Upscale `ESRGAN 2X_Neo_Lite`

![image](https://github.com/vladmandic/automatic/assets/125929259/49f71d57-9b14-41a4-98a9-477b351f1cf8)

Upscale `Latent`

![image](https://github.com/vladmandic/automatic/assets/125929259/faac0bd7-de62-4f77-b52d-9266a165da95)

`Nearest` with a start of `0.8`

![image](https://github.com/vladmandic/automatic/assets/125929259/bd4b7971-4792-4abe-a5e6-7bdf71fc3764)

`ESRGAN 2X_Neo_Lite` with same:

![image](https://github.com/vladmandic/automatic/assets/125929259/6816ed53-c9f8-4281-9407-96627d3002b1)

`Latent`

![image](https://github.com/vladmandic/automatic/assets/125929259/236a673f-ad70-4921-913d-c8010bceeadc)

(better than the alternative, sure)

My previous comments about it only working for non-Latent upscalers were incorrect, and based on weird behavior when using Models that were not "Refiner Specific."  The examples above are using an `SD-XL` Base Model, and `SD-XL Refiner` Refiner Model.

In the end, it looks like you don't want to use Upscale at all in combo with Refiner.  Is that about correct?

**Reply by midcoastal** [reply]
I thought it may have been a `VAE` issue, so I explicitly loaded the `sdxlVAE`, reloaded Refiner and Base, and tried again.

\+ `1man, construction, city, worker`
\- `bad hands, deformed`
`DPM++ 1S`, `50` Steps, CFG `6`, `768x768`

Without Refiner

![image](https://github.com/vladmandic/automatic/assets/125929259/71189baf-c54f-4a5e-83b2-a1c304d81add)

Refiner (`None` Upscaler), Denoise `0.7`, Refiner Start `0.8`, Refiner Steps `5`

![image](https://github.com/vladmandic/automatic/assets/125929259/dfa3d1a3-7500-4a1b-bd3e-3066af04fcc0)

`Nearest` Upscaler (Upscale of `1.5`)

![image](https://github.com/vladmandic/automatic/assets/125929259/790f2be1-6e13-412f-aed8-d903abd44f8b)

`ESRGAN 2X_Neo_Lite`

![image](https://github.com/vladmandic/automatic/assets/125929259/cb249658-a375-4fc0-9e74-a1481f68f26d)

`Latent` Upscaler

![image](https://github.com/vladmandic/automatic/assets/125929259/00de2148-cb11-41ca-8a84-c4fe03328066)

Refiner Steps of `20` and Refiner Start of `0.5`

`None`

![image](https://github.com/vladmandic/automatic/assets/125929259/4382ce45-41ce-410d-a104-bcc0f1169645)

`Nearest`

![image](https://github.com/vladmandic/automatic/assets/125929259/e0ce9d5f-05a2-4736-8d3f-d7adb762c28e)

`ESRGAN 2X_Neo_Lite`

![image](https://github.com/vladmandic/automatic/assets/125929259/0f375a87-801d-47b7-9a4c-6ff6e4ad74cb)

`Latent`

![image](https://github.com/vladmandic/automatic/assets/125929259/e3895b52-ec71-4c8e-bcff-06dbe3ae9b5d)

Seems to me, unless I am sorely missing something, that the Refiner Pipeline takes a snapshot of the Base Render at `Refiner Start` and that gets fed to Secondary, at which point, if you have ANY Upscaler enabled, that processes the snapshot first, and then the Refiner gets it.

This makes the resulting image chaos if using anything other than `Latent` and if using `Latent` the refiner get's an image that was based off the already poor `Latent` upscale of an un-finished image (still full of noise) from the Base Render.  Resulting in...  Yes, some refiner work, but the image is already pretty poor.

For example, this is the `Latent` Upscale with the Refiner unloaded:

![image](https://github.com/vladmandic/automatic/assets/125929259/cb2eddd0-8fe9-4391-8767-459b31e8db85)

Some artifacting, probably from using the incorrect sampler or something, but it looks arguably more "real" than using the Refiner.

I went to go try other Samplers, got one more in after unloading the Refiner, and when I went to try the next one: `sde-dpmsolver++ does is not implemented for <class 'diffusers.schedulers.scheduling_dpmsolver_singlestep.DPMSolverSinglestepScheduler'>`  Odd.

Switched Base Sampler to `UniPC`, as well as Upscaller Sampler.

![image](https://github.com/vladmandic/automatic/assets/125929259/4b06f533-987a-406d-ae9c-e653993ae90f)

Unsure, not really understanding this one all that much...  Shouldn't the Refiner Start be applied to the last render prior to calling the Refiner (IE: `Latent` upscale)?  Or maybe add that much noise back to the image to *imitate* starting at that time-step?  Seeings how all previous renders (Base, and Latent Upscale) all render their full number of steps anyway, it doesn't seem like taking an actual snapshot of the Base render does any good, really.

?  Still wrapping my head around this.  Bare with me.

**Reply by vladmandic** [reply]
all the details do not help at all without knowing base and refiner model used as logic is totally different depending on model type.

and instead of trying absolutely every possible combo, lets think logically and what are we trying to accomplish?
if you're using latent upscale, upscaled image will be unusable without hires. hires uses base model. and after its done, what is the point of handling it over to refiner? what is refiner supposed to do that hires did not do already? could we re-introduce noise once again? sure. but WHY?

and NO, base does not renders full number of steps IF you're using actual refiner model. if you want to use refiner, use actual REFINER model. everything else is nice experimentation, but why are we going into workflows which are more like "just because its possible"?

**Reply by midcoastal** [reply]
> all the details do not help at all without knowing base and refiner model used as logic is totally different depending on model type.

I can't recall the Base Model used, since having success actually triggering what I can identify as "the Refiner Process" (SD.Next recognizes a Model as a Refiner Type, the Base Render gets snapshotted at `0.N` steps, etc) I have only been using `SD XL` Base and Refiners.  Base, obviously something that is not a Refiner, I was using fairly basic Base Models, nothing with baked VAEs, or special render requirements.  The Refiner in all tests was `sdXL_v10Refiner.safetensors`, identified in the UI as a `stable-diffusion-xl-v1-refiner`.

I did change Samplers around, as noted, however.

> and instead of trying absolutely every possible combo, lets think logically and what are we trying to accomplish? if you're using latent upscale, upscaled image will be unusable without hires. hires uses base model. and after its done, what is the point of handling it over to refiner? what is refiner supposed to do that hires did not do already?

Well, that is kind of the question, right?  I am testing every possible combination because I want to see "what" it does, to see "if" I want to or can use it somewhere.  As for "why would you want to upscale with one, and refine with another" for the same reason you would want to do a Base render with one model, and Refine with another, yeah?  Particularly if I have a high `denoise` for the up-scale.  Isn't the point that you use one model for broad-strokes, and another for finer details?

Perhaps you are getting at "why not just use an `ESRGAN` to up-scale," well, that way madness lies, right?

Additionally, why?  Speed.  `SD 1.5` models render at >15it/s for me, and `SD-XL` render at ~8.  So upscale with `SD 1.5` or `2.0` would be faster, then Refine details with `SD-XL`.

And options: I have loads of both `XL` and `non-XL` Models, but at this time, I have more options for Loras, Embeddings, and Models in `non-XL`.  Broad strokes, then details.

But, there is a problem...

> could we re-introduce noise once again? sure. but WHY?

Because snapshotting the Base Render, and pushing that through the pipeline is kind of what this whole thread is on about.

If we forget about taking a snapshot of the Base Render, it makes the "Refiner" process more straight-forward: Do all the things, before Refiner, add the appropriate noise (like "rolling it back to the indicated time-step") then use the Refiner.  Executing the Refiner this way means you can do whatever you want ahead of time, and the Refiner behaves as expected:  Refines the Image using `X%` of the original image (considering that a `start` of, say `0.6` is the same as saying "Refine the image after the Base has done 60% of the diffusion").  IIUC: This is kind of the same as how `denoise` works, yeah?

It would be a different story if the Base diffusion actually broke at `0.6` or whatever, but it doesn't.

Another reason to do it the "just add back the noise" way relates the the problem above: It doesn't look to me like you can mix and match `XL` and `non-XL` Models when using Refiners (`non-XL` Base, and `XL` Refiner).

Without any Upscaling, Base of `copaxRealistic_v2`, prompt `portrait, close-up, 1man, construction, worker, 4k, absurdres, editorial photograph, city`:

![image](https://github.com/vladmandic/automatic/assets/125929259/c913e7e3-18bd-4c7b-8762-e886a189af30)

No upscaling (Upscaler `None`), Refiner of `sdXL_v10RefinerVAEFix`, Start of `0.8`, Steps of `10`:

![image](https://github.com/vladmandic/automatic/assets/125929259/a1634ee8-1a6e-4e18-b698-e1f9f84ead7c)

Maybe start earlier, give the Refiner more to do? Start `0.5`:

![image](https://github.com/vladmandic/automatic/assets/125929259/ada3e72c-ff9d-4d09-a387-d8371627295c)

Maybe it is the Sampler (even though I have been using this Sampler with the `XL` models all day), I tried all of them, nothing better.

Maybe it is the Refiner (I am using the `VAEFix` version because I don't want to load a VAE that is incompatible with either Model), switching to the non-`VAEFix` version of this same model was almost the same:

![image](https://github.com/vladmandic/automatic/assets/125929259/bedd04d3-93e4-495e-941e-cbdfa6688e12)

What I found interesting about **this** experiment was that the Refiner wasn't executing in full:

```
Progress  3.85it/s ████████████████▋                54% 7/13 00:01 00:01 Refiner
```

Not only that, but my refiner steps were set at `10`.  It was showingf `3` more than set, and executing `3 ` LESS than was set...  Changing to `20` got me:

```
Progress  6.77it/s ██████████████████████████████ 100% 11/11 00:01 00:00 Refiner
```

Ok...  And changing back to `10` got me.....

```
Progress  5.38it/s ████████████████████████████████ 100% 6/6 00:01 00:00 Refiner
```

Huh?? Changing to `5` got me `3` steps, and changing back to `10` got me to `6` again...  Oddly, the only setting that seemed to change the number of steps the Refiner took was the Refiner Steps setting it's self, nothing else effected it (as I assumed would, since it is not doing the number of steps set).  Not either CFG, Guidance, the Base Steps, the Denoise, the Refiner Start, nothing changed that number.  Weird...

Anyway, another tangent, the point is: By doing the "just add the required Noise to image X", mixing and matching anything would just work (I am probably incorrect assuming that the incompatibility between `SD` Models and `SD-XL` Refiners is the different VAE decoding, like, the numerical representation of the data getting fed to the Refiner is not in the same space as would be an SD-XL model [same reason `VAEs` are bound to each type]).

Just do all the stuff, make a full image, encode into Refiner space, add the noise, time-step out of it and good to go.  That's why.

Because, let's be honest, ultimately all Refiners are is I2I with sugar, right?  If not, what is the practical difference (in light of the fact that it does, ultimately, render the whole pipeline, regardless)?

On that.

> and NO, base does not renders full number of steps IF you're using actual refiner model. if you want to use refiner, use actual REFINER model. everything else is nice experimentation, but why are we going into workflows which are more like "just because its possible"?

Are you sure?  

To double-check, I switched to `dev`, moved my stash of models so that I only had a handful of SD Models to chose from (and very few Embeddings/Loras/etc, partially to limit influence, mostly so that loading would be faster on `dev`), ran a `git pull` with `--recurse-submodules`, and launched with `--reinstall` (I did all of this because I was getting errors about `diffusers` not having `Kandinsky 3`, `AttributeError: 'NoneType' object has no attribute 'named_modules'` in the Lora Extension, and `Model detected as SD-XL refiner model, but attempting to load a base model` when explicitly trying to load a Refiner Model).

Here are my logs (cleaned) from the first Model Load to finish:

```
INFO     Select: model="sdXL_v10Refiner/sdXL_v10Refiner"                                                                                                                                                                                 
DEBUG    Load model weights: existing=False target={webui_path}/models/Stable-diffusion/sdXL_v10Refiner/sdXL_v10Refiner.safetensors info=None                                                           
DEBUG    Unload weights model: {'ram': {'used': 17.56, 'total': 125.53}, 'gpu': {'used': 8.11, 'total': 11.71}, 'retries': 0, 'oom': 0}                                                                                                  
DEBUG    gc: collected=613 device=cuda {'ram': {'used': 17.56, 'total': 125.53}, 'gpu': {'used': 1.14, 'total': 11.71}, 'retries': 0, 'oom': 0}                                                                                          
Loading model: {webui_path}/models/Stable-diffusion/sdXL_v10Refiner/sdXL_v10Refiner.safetensors ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 0.0/6.1 GB -:--:--
DEBUG    Server: alive=True jobs=0 requests=162 uptime=137 memory=23.06/125.53 backend=Backend.DIFFUSERS state=idle                                                                                                                      
Loading model: {webui_path}/models/Stable-diffusion/sdXL_v10Refiner/sdXL_v10Refiner.safetensors ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.1/6.1 GB 0:00:00
DEBUG    Desired Torch parameters: dtype=FP16 no-half=False no-half-vae=False upscast=False                                                                                                                                              
INFO     Setting Torch parameters: device=cuda dtype=torch.float16 vae=torch.float16 unet=torch.float16 context=no_grad fp16=True bf16=False                                                                                             
WARNING  VAE not found: sdxl_vae.safetensors                                                                                                                                                                                             
WARNING  Model detected as SD-XL refiner model, but attempting to load a base model: model={webui_path}/models/Stable-diffusion/sdXL_v10Refiner/sdXL_v10Refiner.safetensors size=5795 MB                
INFO     Autodetect: model="Stable Diffusion XL" class=StableDiffusionXLPipeline file="{webui_path}/models/Stable-diffusion/sdXL_v10Refiner/sdXL_v10Refiner.safetensors" size=5795MB                    
DEBUG    Setting model: pipeline=StableDiffusionXLImg2ImgPipeline config={'low_cpu_mem_usage': True, 'torch_dtype': torch.float16, 'load_connected_pipeline': True, 'variant': 'fp16', 'extract_ema': True, 'force_zeros_for_empty_prompt
         ': True, 'requires_aesthetics_score': False, 'use_safetensors': True}                                                                                                                                                           
DEBUG    Setting model: enable VAE slicing                                                                                                                                                                                               
DEBUG    Setting model: enable VAE tiling                                                                                                                                                                                                
DEBUG    Setting model VAE: name=None upcast=True                                                                                                                                                                                        
INFO     Load embeddings: loaded=0 skipped=1866 time=0.08                                                                                                                                                                                
DEBUG    gc: collected=659 device=cuda {'ram': {'used': 18.44, 'total': 125.53}, 'gpu': {'used': 6.94, 'total': 11.71}, 'retries': 0, 'oom': 0}                                                                                          
ERROR    executing callback: {webui_path}/extensions-builtin/Lora/scripts/lora_script.py model_loaded_callback: AttributeError                                                                          
╭───────────────────────────────────────────────────────────────────────────────── Traceback (most recent call last) ──────────────────────────────────────────────────────────────────────────────────╮
│ {webui_path}/modules/script_callbacks.py:173 in model_loaded_callback                                                                                               │
│                                                                                                                                                                                                      │
│   172 │   │   │   t0 = time.time()                                                                                                                                                                   │
│ ❱ 173 │   │   │   c.callback(sd_model)                                                                                                                                                               │
│   174 │   │   │   timer(t0, c.script, 'model_loaded')                                                                                                                                                │
│                                                                                                                                                                                                      │
│ {webui_path}/extensions-builtin/Lora/networks.py:49 in assign_network_names_to_compvis_modules                                                                      │
│                                                                                                                                                                                                      │
│    48 │   │   │   return                                                                                                                                                                             │
│ ❱  49 │   │   for name, module in shared.sd_model.text_encoder.named_modules():                                                                                                                      │
│    50 │   │   │   prefix = "lora_te1_" if shared.sd_model_type == "sdxl" else "lora_te_"                                                                                                             │
╰──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
AttributeError: 'NoneType' object has no attribute 'named_modules'
INFO     Load model: time=8.61 { load=8.61 } native=1024 {'ram': {'used': 18.44, 'total': 125.53}, 'gpu': {'used': 6.94, 'total': 11.71}, 'retries': 0, 'oom': 0}                                                                        
DEBUG    Save: file="config.json" json=36 bytes=1609                                                                                                                                                                                     
DEBUG    Unused settings: ['multiple_tqdm']                                                                                                                                                                                              
DEBUG    Setting changed: key=sd_model_checkpoint, value=sdXL_v10Refiner/sdXL_v10Refiner                                                                                                                                                 
INFO     Select: model="copaxTimelessxlSDXL1_v42/copaxTimelessxlSDXL1_v42"                                                                                                                                                               
DEBUG    Load model weights: existing=False target={webui_path}/models/Stable-diffusion/copaxTimelessxlSDXL1_v42/copaxTimelessxlSDXL1_v42.safetensors info=None                                         
DEBUG    Unload weights model: {'ram': {'used': 18.46, 'total': 125.53}, 'gpu': {'used': 6.93, 'total': 11.71}, 'retries': 0, 'oom': 0}                                                                                                  
DEBUG    gc: collected=2109 device=cuda {'ram': {'used': 17.94, 'total': 125.53}, 'gpu': {'used': 1.14, 'total': 11.71}, 'retries': 0, 'oom': 0}                                                                                         
Loading model: {webui_path}/models/Stable-diffusion/copaxTimelessxlSDXL1_v42/copaxTimelessxlSDXL1_v42.safetensors ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.9/6.9 GB 0:00:00
DEBUG    Desired Torch parameters: dtype=FP16 no-half=False no-half-vae=False upscast=False                                                                                                                                              
INFO     Setting Torch parameters: device=cuda dtype=torch.float16 vae=torch.float16 unet=torch.float16 context=no_grad fp16=True bf16=False                                                                                             
WARNING  VAE not found: sdxl_vae.safetensors                                                                                                                                                                                             
INFO     Autodetect: model="Stable Diffusion XL" class=StableDiffusionXLPipeline file="{webui_path}/models/Stable-diffusion/copaxTimelessxlSDXL1_v42/copaxTimelessxlSDXL1_v42.safetensors" size=6617MB  
DEBUG    Setting model: pipeline=StableDiffusionXLPipeline config={'low_cpu_mem_usage': True, 'torch_dtype': torch.float16, 'load_connected_pipeline': True, 'variant': 'fp16', 'extract_ema': True, 'force_zeros_for_empty_prompt ':    
         True, 'requires_aesthetics_score': False, 'use_safetensors': True}                                                                                                                                                              
DEBUG    Setting model: enable VAE slicing                                                                                                                                                                                               
DEBUG    Setting model: enable VAE tiling                                                                                                                                                                                                
DEBUG    Setting model VAE: name=None upcast=True                                                                                                                                                                                        
INFO     Load embeddings: loaded=13 skipped=1853 time=0.27                                                                                                                                                                               
DEBUG    gc: collected=863 device=cuda {'ram': {'used': 20.61, 'total': 125.53}, 'gpu': {'used': 8.15, 'total': 11.71}, 'retries': 0, 'oom': 0}                                                                                          
INFO     Load model: time=4.69 { load=4.69 } native=1024 {'ram': {'used': 20.61, 'total': 125.53}, 'gpu': {'used': 8.15, 'total': 11.71}, 'retries': 0, 'oom': 0}                                                                        
DEBUG    Save: file="config.json" json=36 bytes=1627                                                                                                                                                                                     
DEBUG    Unused settings: ['multiple_tqdm']                                                                                                                                                                                              
DEBUG    Setting changed: key=sd_model_checkpoint, value=copaxTimelessxlSDXL1_v42/copaxTimelessxlSDXL1_v42                                                                                                                               
DEBUG    txt2img: id_task=task(0p55b0qasjmhu02)|prompt=1man, construction, city, worker|negative_prompt=bad hands,                                                                                                                       
         deformed|prompt_styles=[]|steps=20|sampler_index=1|latent_index=1|full_quality=True|restore_faces=False|tiling=False|n_iter=1|batch_size=1|cfg_scale=6|clip_skip=1|seed=-1.0|subseed=-1.0|subseed_strength=0|seed_resize_from_h=
         0|seed_resize_from_w=0||height=768|width=512|enable_hr=True|denoising_strength=0.7|hr_scale=1.5|hr_upscaler=Latent|hr_force=False|hr_second_pass_steps=30|hr_resize_x=0|hr_resize_y=0|image_cfg_scale=6|diffusers_guidance_resca
         le=0.7|refiner_steps=5|refiner_start=0.8|refiner_prompt=|refiner_negative=|override_settings_texts=[]                                                                                                                           
INFO     Select: refiner="sdXL_v10Refiner/sdXL_v10Refiner"                                                                                                                                                                               
DEBUG    Load model weights: existing=False target={webui_path}/models/Stable-diffusion/sdXL_v10Refiner/sdXL_v10Refiner.safetensors info=None                                                           
Loading model: {webui_path}/models/Stable-diffusion/sdXL_v10Refiner/sdXL_v10Refiner.safetensors ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.1/6.1 GB 0:00:00
DEBUG    Desired Torch parameters: dtype=FP16 no-half=False no-half-vae=False upscast=False                                                                                                                                              
INFO     Setting Torch parameters: device=cuda dtype=torch.float16 vae=torch.float16 unet=torch.float16 context=no_grad fp16=True bf16=False                                                                                             
WARNING  VAE not found: sdxl_vae.safetensors                                                                                                                                                                                             
INFO     Autodetect: refiner="Stable Diffusion XL" class=StableDiffusionXLPipeline file="{webui_path}/models/Stable-diffusion/sdXL_v10Refiner/sdXL_v10Refiner.safetensors" size=5795MB                  
DEBUG    Setting refiner: pipeline=StableDiffusionXLImg2ImgPipeline config={'low_cpu_mem_usage': True, 'torch_dtype': torch.float16, 'load_connected_pipeline': True, 'variant': 'fp16', 'extract_ema': True,                            
         'force_zeros_for_empty_prompt ': True, 'requires_aesthetics_score': False, 'use_safetensors': True}                                                                                                                             
DEBUG    Setting refiner: enable VAE slicing                                                                                                                                                                                             
DEBUG    Setting refiner: enable VAE tiling                                                                                                                                                                                              
DEBUG    Setting refiner VAE: name=None upcast=True                                                                                                                                                                                      
DEBUG    Moving refiner model to CPU                                                                                                                                                                                                     
INFO     Load embeddings: loaded=13 skipped=1853 time=0.16                                                                                                                                                                               
DEBUG    gc: collected=1132 device=cuda {'ram': {'used': 33.72, 'total': 125.53}, 'gpu': {'used': 8.12, 'total': 11.71}, 'retries': 0, 'oom': 0}                                                                                         
INFO     Load refiner: time=4.75 { load=4.75 } native=1024 {'ram': {'used': 33.72, 'total': 125.53}, 'gpu': {'used': 8.12, 'total': 11.71}, 'retries': 0, 'oom': 0}                                                                      
DEBUG    Diffuser pipeline: StableDiffusionXLPipeline task=DiffusersTaskType.TEXT_2_IMAGE set={'prompt_embeds': torch.Size([1, 77, 2048]), 'pooled_prompt_embeds': torch.Size([1, 1280]), 'negative_prompt_embeds': torch.Size([1, 77,   
         2048]), 'negative_pooled_prompt_embeds': torch.Size([1, 1280]), 'guidance_scale': 6, 'generator': device(type='cuda'), 'output_type': 'latent', 'num_inference_steps': 25, 'eta': 1.0, 'guidance_rescale': 0.7, 'denoising_end':
         0.8, 'height': 768, 'width': 512, 'parser': 'Full parser'}                                                                                                                                                                      
DEBUG    Sampler: sampler="UniPC" config={'num_train_timesteps': 1000, 'beta_start': 0.00085, 'beta_end': 0.012, 'beta_schedule': 'scaled_linear', 'prediction_type': 'epsilon', 'solver_order': 2, 'thresholding': False,               
         'sample_max_value': 1.0, 'predict_x0': 'bh1', 'lower_order_final': False}                                                                                                                                                       
Progress  8.19it/s █████████████████████████████████ 100% 21/21 00:02 00:00 Base
DEBUG    Init hires: upscaler="Latent" sampler="UniPC" resize=0x0 upscale=768x1152                                                                                                                                                       
INFO     Hires: upscaler=Latent width=768 height=1152 images=1                                                                                                                                                                           
DEBUG    Moving to CPU: model=UNet                                                                                                                                                                                                       
DEBUG    VAE decode: name=baked dtype=torch.float32 upcast=True images=1 latents=torch.Size([1, 4, 144, 96]) time=5.125                                                                                                                  
DEBUG    Pipeline class change: original=StableDiffusionXLPipeline target=StableDiffusionXLImg2ImgPipeline                                                                                                                               
DEBUG    Sampler: sampler="UniPC" config={'num_train_timesteps': 1000, 'beta_start': 0.00085, 'beta_end': 0.012, 'beta_schedule': 'scaled_linear', 'prediction_type': 'epsilon', 'solver_order': 2, 'thresholding': False,               
         'sample_max_value': 1.0, 'predict_x0': 'bh1', 'lower_order_final': False}                                                                                                                                                       
DEBUG    Diffuser pipeline: StableDiffusionXLImg2ImgPipeline task=DiffusersTaskType.IMAGE_2_IMAGE set={'prompt_embeds': torch.Size([1, 77, 2048]), 'pooled_prompt_embeds': torch.Size([1, 1280]), 'negative_prompt_embeds':              
         torch.Size([1, 77, 2048]), 'negative_pooled_prompt_embeds': torch.Size([1, 1280]), 'guidance_scale': 6, 'generator': device(type='cuda'), 'output_type': 'latent', 'num_inference_steps': 43, 'eta': 1.0, 'guidance_rescale':   
         0.7, 'image': <class 'list'>, 'strength': 0.7, 'parser': 'Full parser'}                                                                                                                                                         
Progress  4.21it/s ████████████████████████████████ 100% 30/30 00:07 00:00 Hires
DEBUG    Moving to CPU: model=base                                                                                                                                                                                                       
DEBUG    Sampler: sampler="UniPC" config={'num_train_timesteps': 1000, 'beta_start': 0.00085, 'beta_end': 0.012, 'beta_schedule': 'scaled_linear', 'prediction_type': 'epsilon', 'solver_order': 2, 'thresholding': False,               
         'sample_max_value': 1.0, 'predict_x0': 'bh1', 'lower_order_final': False}                                                                                                                                                       
DEBUG    Pipeline class change: original=StableDiffusionXLImg2ImgPipeline target=StableDiffusionXLPipeline                                                                                                                               
DEBUG    Diffuser pipeline: StableDiffusionXLImg2ImgPipeline task=DiffusersTaskType.IMAGE_2_IMAGE set={'prompt_embeds': torch.Size([1, 77, 1280]), 'pooled_prompt_embeds': torch.Size([1, 1280]), 'negative_prompt_embeds':              
         torch.Size([1, 77, 1280]), 'negative_pooled_prompt_embeds': torch.Size([1, 1280]), 'guidance_scale': 6, 'generator': device(type='cuda'), 'output_type': 'latent', 'num_inference_steps': 13, 'eta': 1.0, 'guidance_rescale':   
         0.7, 'denoising_start': 0.8, 'denoising_end': 1, 'image': <class 'torch.Tensor'>, 'parser': 'Full parser'}                                                                                                                      
Progress  1.52it/s ████████████████████████████████ 100% 2/2 00:01 00:00 Refiner
DEBUG    Moving to CPU: model=UNet                                                                                                                                                                                                       
DEBUG    VAE decode: name=baked dtype=torch.float32 upcast=True images=1 latents=torch.Size([1, 4, 144, 96]) time=0.588                                                                                                                  
DEBUG    Moving to CPU: model=refiner                                                                                                                                                                                                    
INFO     GPU high memory utilization: 96% {'ram': {'used': 34.4, 'total': 125.53}, 'gpu': {'used': 11.19, 'total': 11.71}, 'retries': 1, 'oom': 0}                                                                                       
DEBUG    gc: collected=296 device=cuda {'ram': {'used': 34.4, 'total': 125.53}, 'gpu': {'used': 1.98, 'total': 11.71}, 'retries': 1, 'oom': 0}                                                                                           
DEBUG    Saving: image="outputs/text/07345-copaxTimelessxlSDXL1_v42-1man construction city worker.jpg" type=JPEG size=768x1152                                                                                                           
INFO     Processed: images=1 time=26.08 its=0.77 memory={'ram': {'used': 34.4, 'total': 125.53}, 'gpu': {'used': 2.51, 'total': 11.71}, 'retries': 1, 'oom': 0} 
```

(Oops on the VAE, I forgot to copy it over, I did, and re-tested, same result, execution-wise)

I can confirm that, while watching the WebUI, and the console, the WebUI Preview "stops" at the indicated time-step, however, the console continues rendering (the progress indicator continues) for the remaining steps (On the `Base` render).

**Reply by midcoastal** [reply]
I also went and gave the Refiner more Steps, take it from me that 30 didn't show a difference, neither did 50, I can't imagine 100 would be different, and 5, about the same.  Bump CFG to 10?  Nope.

Maybe I should note that is is with basically all the default `Settings`:

![image](https://github.com/vladmandic/automatic/assets/125929259/9ccdebab-cef6-4c08-ae43-778707f0650d)
![image](https://github.com/vladmandic/automatic/assets/125929259/620cdab0-693e-4477-a0a8-1794e09625f4)
![image](https://github.com/vladmandic/automatic/assets/125929259/43171f41-258e-476c-ae9b-3170605b6720)

Incorrect?

**Reply by midcoastal** [reply]
Double-checking that it is an `SD` vs `SD-XL` issue, I tried `copaxTimelessxlSDXL1_v6` as a Base:

No Refiner:

![image](https://github.com/vladmandic/automatic/assets/125929259/151c366b-5779-455c-ae74-682f5d899ec2)

Refiner with `0.8` and `10`:

![image](https://github.com/vladmandic/automatic/assets/125929259/dfe33ad4-e764-442c-82e4-49d376a8c104)

Refiner with `0.5` and `30`:

![image](https://github.com/vladmandic/automatic/assets/125929259/fee28695-4bb2-4e74-8f55-89f5e1ebdb23)

It's great!  So, definitely a Model conflict.
